src/axolotl/utils/config.py

```diff
@@ -78,9 +78,13 @@ def normalize_config(cfg):
         cfg.torch_dtype = torch.float32
 
     if cfg.saves_per_epoch:
-        cfg.save_steps = 1.0 / (cfg.saves_per_epoch * cfg.num_epochs)
+        save_steps = 1.0 / (cfg.saves_per_epoch * cfg.num_epochs)
+        if save_steps < 1.0:  # prevent saves on every step
```

Comment:
maybe instead of this `if save_steps < 1.0:`, you should do a check that `cfg.saves_per_epoch >= 1`